jdaviz/core/unit_conversion_utils.py

```diff
@@ -0,0 +1,398 @@
+    if spectral_axis_unit in (u.pix, u.dimensionless_unscaled):
+        return [spectral_axis_unit]
+
+    # Get unit equivalencies.
+    try:
+        curr_spectral_axis_unit_equivalencies = spectral_axis_unit.find_equivalent_units(
+            equivalencies=u.spectral())
+    except u.core.UnitConversionError:
+        return []
+
+    # Get local units.
+    locally_defined_spectral_axis_units = ['Angstrom', 'nm',
+                                           'um', 'Hz', 'erg']
+    local_units = [u.Unit(unit) for unit in locally_defined_spectral_axis_units]
+
+    # Remove overlap units.
+    curr_spectral_axis_unit_equivalencies = list(set(curr_spectral_axis_unit_equivalencies)
+                                                 - set(local_units + exclude))
+
+    # Convert equivalencies into readable versions of the units and sorted alphabetically.
+    spectral_axis_unit_equivalencies_titles = sorted(units_to_strings(
+        curr_spectral_axis_unit_equivalencies))
+
+    # Concatenate both lists with the local units coming first.
+    return sorted(units_to_strings(local_units)) + spectral_axis_unit_equivalencies_titles
+
+
+def flux_conversion_general(values, original_unit, target_unit,
+                            equivalencies=None, with_unit=True):
+    """
+    Converts ``values`` from ``original_unit`` to ``target_unit`` using the
+    provided ``equivalencies`` while handling special cases where direct unit
+    conversion is not possible. This function is designed to account for
+    scenarios like conversions involving flux to surface brightness that
+    also require a ``u.spectral_density`` equivalency, conversions between
+    per-square pixel surface brightnesses that don't convert directly, and other
+    flux to surface brightness conversions.
+
+    This function should be used for unit conversions when possible instead of
+    directly using Astropy's ``unit.to()``, as it handles additional logic for
+    special cases.
+
+    Parameters
+    ----------
+    values : array-like or float
+        The numerical values to be converted.
+    original_unit : ``astropy.units.Unit`` or str
+        The unit of the input values.
+    target_unit : ``astropy.units.Unit`` or str
+        The desired unit to convert to.
+    equivalencies : list of equivalencies, optional
+        Unit equivalencies to apply during the conversion.
+    with_unit : bool, optional
+        If True, the returned value retains its unit. If False, only the
+        numerical values are returned.
+
+    Returns
+    -------
+    converted_values : Quantity or float
+        The converted values, with or without units based on ``with_unit``.
+
+    Raises
+    ------
+    `astropy.units.UnitConversionError`
+        If the conversion between ``original_unit`` and ``target_unit`` fails
+        despite the provided equivalencies.
+
+    """
+
+    if original_unit == target_unit:
+        if not with_unit:
+            return (values * original_unit).value
```

Comment:
Why not just return the input values here? 
```suggestion
            return values
```